app/models/file_depot_ftp.rb

```diff
@@ -93,12 +93,15 @@ def file_exists?(file_or_directory)
   private
 
   def create_directory_structure(directory_path)
-    Pathname.new(directory_path).descend do |path|
```

Comment:
I see. The new code drills down the directory structure by `mkdir A; chdir A` the same way. So the IIS's inability to create directories like `mkdir -p` should not be a problem. However, I will try to get some IIS for testing.